src/Symfony/Component/Security/Http/Authenticator/AuthenticatorInterface.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\Authenticator;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+/**
+ * The interface for all authenticators.
+ *
+ * @author Ryan Weaver <ryan@knpuniversity.com>
+ * @author Amaury Leroux de Lens <amaury@lerouxdelens.com>
+ * @author Wouter de Jong <wouter@wouterj.nl>
+ *
+ * @experimental in 5.1
+ */
+interface AuthenticatorInterface
```

Comment:
Idea: rename this `AuthenticatorStrategyInterface` and rename `AuthenticatorManagerInterface` to `AuthenticatorInterface`?

Reply:
Oh, I like the idea of renaming this interface to free up with `Authenticator` name. There is 2 things I need to think about a bit more:
- I'm not sure about "strategy", but I really explain why yet. I'll see what other frameworks/docs use for this.
- We have to be aware that this interface is probably used more than the manager interface. So we should try to also keep this one not too long imho

Reply:
> this interface is probably used more than the manager interface

I was that there is an autowiring alias for `AuthenticatorManagerInterface`. If that's a common need (injecting this service), then I'm not sure which interface is used more than the other. Injecting a service looks potentially more frequent than implementing an interface to me.